### ISSUE TITLE
How to setup ? :D

### ISSUE THREAD

**Issue by At0micA55** [merlinlikethewizard/Mastermine#1]
Hello ! First I would like to thank you a lot for the hard work you put into that ! Like it is wayyyyy cooler then i tough it would be just looking at the videos. When i see all the files i am amazed on how you did this !!!! 

But I got some questions about it... 
In your video you say there is a config file but i can't seem to find it :/ I know you say the system is not *yet* users friendly but i think that a tutorial on how to setup the system would be really appreciated ! (Looking at the youtube comments i see a lot of questions like "how to set it up ?") And i can see that you are more active here then on youtube so this is why i am asking this here ^^

PS: I would gladly make a tutorial for you if you wish once I understand how to set it up ^^

**Comment by merlinlikethewizard** [comment]
Hi! Thank you so much!! And yes, I definitely think it's worth it to make this thing actually usable now that people are taking an interest.

The main necessary change is to make coordinates relative rather than absolute. The config file I talked about is actually just a .lua file that gets read in as a module, it's the hub_files/config.lua, and in it are the routes the turtles take to get between their home, the item dropoff, the fuel dropoff/pickup, and the mine, which is actually a loop to prevent collisions. If all of those coordinates were relative to the "mine_enter" coordinate then all that you'd need to start a new mine would be to change that field. That's not a difficult fix so maybe I'll do that today.

I was thinking I would make a tutorial pretty soon, I do want to change a few more things like the way the screen works (you can only have one, and the top of the map is set to south), but I would say in the next week or so I'll get a tutorial up. I feel bad taking a while getting to it, I hate it when people show off things without tutorials :/

Anyway, I needed a nudge so thanks! Let me know if you have any more questions or suggestions :)

**Comment by PracticalYT** [comment]
This thing is beast. I would love this thing to be easily deployable! But for now I'll try my best to get it working.

**Comment by Eforen** [comment]
I am also unsure how to set this up in CC: T on 1.16 because I can't seem to find a mod that does chunk loading. What do you use @merlinlikethewizard or do you not use 1.16?

**Comment by merlinlikethewizard** [comment]
I'm still using 1.12 with the PeripheralsPlusOne mod which isn't on curseforge but seems to still be available on github

https://github.com/rolandoislas/PeripheralsPlusOne/releases/latest

Unfortunately, I'm pretty sure there's nothing like it available for 1.16. I'm really hoping someone will make a port at some point, but as far as I know it hasn't happened yet. This is why it would be really great for a version of the system that didn't need the chunk loaders, I wasn't able to come up with a way to do it in the base mod but maybe someone will think of something.

**Comment by RubyTemple** [comment]
but how i can put this project on floppy disk?

**Comment by merlinlikethewizard** [comment]
The easiest way is to copy the code directly to the disk folder. First you want to craft a floppy disk and initialize it by making a folder on it or editing a file, just to get the game to recognize the disk. Then find your world save, and locate the directory for the disk which should be <savename>/computer/disk/<disk#>/ and copy all the code into that directory.

**Comment by lazy-engineer100** [comment]
how would i download the files onto a floppy drive if i do not have access to the world file? (like if im playing on a server someone else is hosting)
i was able to download the zip file, but i do not know how to extract files from the zip, if it is even at all possible

**Comment by brettrm001** [comment]
You could ask the host to do it for you, or you could upload each file to pastebin, and download it to the floppy disk (Make sure you create the sub folders like hub_files and name everything the exact same.)

**Comment by merlinlikethewizard** [comment]
Yeah I didn't really think of this until now cuz I've been on single player, though it seems pretty important huh. I guess whole directories can't be uploaded to pastebin, but maybe there's a cheaty way to concat all the files into one big one and then have the turtle build itself after downloading... Maybe I'll look into that.

But yes, for now @brettrm001 is right, those would be the options.

**Comment by PracticalYT** [comment]
> Yeah I didn't really think of this until now cuz I've been on single player, though it seems pretty important huh. I guess whole directories can't be uploaded to pastebin, but maybe there's a cheaty way to concat all the files into one big one and then have the turtle build itself after downloading... Maybe I'll look into that.
> 
> But yes, for now @brettrm001 is right, those would be the options.

Maybe we can use the "wget" command on a computer to download the folder onto a floppy disk, ideally this file would be ready to go, so maybe have a setup script that has the player enter the xyz coordinates, the program then inserts those variables into the files or into an external variable file where all the files just pull variables from, much like in javascript you can pull variables out of an external .json file. If you can make this possible, that would be amazing haha. Maybe you could generate a bit of income out our interest like putting this download on your own website and generate income from google ads.

**Comment by brettrm001** [comment]
If he wanted to he could probably make a setup script but there really isn't a need for it. It is pretty simple to edit the start coordinate of the mine and what level you want it to mine on.

**Comment by PracticalYT** [comment]
Yup!

**Comment by merlinlikethewizard** [comment]
@PracticalYT That definitely sounds like the "right" way to do it, as an actual download and everything. Probably that will be the route to go in the future (I like the monetization idea too), but for now I did the one that was easier though maybe a little insane. It's all in one file!

@lazy-engineer100 this should solve your problem.

Here's the new instructions (which I'll put in the readme in a sec):

1. Place your advanced computer next to a disk drive with a blank disk in.
2. Run "pastebin get Wu3LRF99 mastermine.lua"
3. Run "mastermine disk"
4. Run "disk/hub.lua"

And for anyone interested in the code I just wrote to put it all in one file, here ya go:
"pastebin get sMN9bCPt conglomerate.lua"
